pkg/neonvm/cpuscaling/sysfs.go

```diff
@@ -0,0 +1,127 @@
+package cpuscaling
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"slices"
+	"strconv"
+	"strings"
+)
+
+// CPU directory path
+const cpuPath = "/sys/devices/system/cpu/"
+
+type cpuSysfsState struct{}
+
+func (cs *cpuSysfsState) SetState(cpuNum int, cpuState cpuState) error {
+	var state string
+	switch cpuState {
+	case cpuOnline:
+		state = "1"
+	case cpuOffline:
+		state = "0"
+	}
+
+	err := os.WriteFile(filepath.Join(cpuPath, fmt.Sprintf("cpu%d/online", cpuNum)), []byte(state), 0o644)
+	if err != nil {
+		return fmt.Errorf("failed to set CPU %d online status: %w", cpuNum, err)
+	}
+
+	return nil
+}
+
+func (cs *cpuSysfsState) GetState(cpuNum int) (cpuState, error) {
+	data, err := os.ReadFile(filepath.Join(cpuPath, "online"))
+	if err != nil {
+		return cpuOffline, err
+	}
+
+	onlineCPUs, err := cs.parseMultipleCPURange(string(data))
+	if err != nil {
+		return cpuOffline, err
+	}
+
+	if slices.Contains(onlineCPUs, cpuNum) {
+		return cpuOnline, nil
+	}
+
+	return cpuOffline, nil
+}
+
+// PossibleCPUs returns the start and end indexes of all possible CPUs.
+func (cs *cpuSysfsState) PossibleCPUs() (int, int, error) {
+	data, err := os.ReadFile(filepath.Join(cpuPath, "possible"))
+	if err != nil {
+		return -1, -1, err
+	}
+
+	return cs.parseCPURange(string(data))
+}
+
+// parseCPURange parses the CPU range string (e.g., "0-3") and returns a list of CPUs.
```

Comment:
> returns a list of CPUs.

question: This is no longer true, right? Let's change it?

Reply:
changed